### ISSUE TITLE
multi-class detection vs few-class detection ?

### ISSUE THREAD

**Issue by hoangkhoiLE** [ultralytics/yolov5#5170]
## ❔Question
Dear @glenn-jocher,
I want to do detection on person and bag object which are two class from COCO dataset. So, if i want to get a model better performance in only person and bag, do i need to retrain a model with label filtered between that two class and remove label of other classes ? Can you give me an advise in this situation ? Thank in advance !

Best regard !

**Comment by glenn-jocher** [comment]
@hoangkhoiLE yes the fewer classes the model trains on the better it will perform at each class naturally. I've created PR https://github.com/ultralytics/yolov5/pull/5172 to allow for class filtering by manually updating datasets.py.  

To train on person and bag classes only set this variable in datasets.py accordingly:
```python
        include_class = [0, 26]  # filter labels to include only these classes (optional)
```

**Comment by glenn-jocher** [comment]
@hoangkhoiLE good news 😃! Your original issue may now be fixed ✅ in PR #5172 which is now merged. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by MattVAD** [comment]
Hello @glenn-jocher, thank you for fixing this ! I'll try it soon. Just one more question, in your opinion/experience, if I want to train a model on only 2 classes (with the example of the amazing [hoangkhoiLE](https://github.com/hoangkhoiLE), person & bag), is it better to use the pretrained weights or train the model from scratch ?

**Comment by glenn-jocher** [comment]
@MattVAD for most use cases (smaller datasets) starting from pretrained models will deliver better results. If you have large (COCO-size) dataset, then training from scratch may also produce good results.

I would train both ways if I were you to compare.
